filters/ratelimit/ratelimit.go

```diff
@@ -343,13 +379,30 @@ func getDurationArg(a interface{}) (time.Duration, error) {
 	return time.Duration(i) * time.Second, err
 }
 
-// Request stores the configured ratelimit.Settings in the state bag,
-// such that it can be used in the proxy to check ratelimit.
+// Request checks ratelimit using filter settings and serves `429 Too Many Requests` response if limit is reached
 func (f *filter) Request(ctx filters.FilterContext) {
-	if settings, ok := ctx.StateBag()[RouteSettingsKey].([]ratelimit.Settings); ok {
-		ctx.StateBag()[RouteSettingsKey] = append(settings, f.settings)
-	} else {
-		ctx.StateBag()[RouteSettingsKey] = []ratelimit.Settings{f.settings}
+	rateLimiter := f.provider.get(f.settings)
+	if rateLimiter == nil {
+		log.Errorf("RateLimiter is nil for settings: %s", f.settings)
+		return
+	}
+
+	if f.settings.Lookuper == nil {
+		log.Errorf("Lookuper is nil for settings: %s", f.settings)
+		return
+	}
+
+	s := f.settings.Lookuper.Lookup(ctx.Request())
+	if s == "" {
+		log.Debugf("Lookuper found no data in request for settings: %s and request: %v", f.settings, ctx.Request())
+		return
+	}
+
+	if !rateLimiter.AllowContext(ctx.Request().Context(), s) {
+		ctx.Serve(&http.Response{
```

Comment:
There is no body here. Global ratelimiter uses [http.Error](https://golang.org/pkg/net/http/#Error) that sends text/plain body.